### DISCUSSION TITLE
Rocket League game files not detected after reboot

### DISCUSSION THREAD

**Discussion by GetPsyched** [fufexan/nix-gaming#75] [Q&A] [answered]
For context, I have impermanence enabled on NixOS. However, before rebooting, I had persisted `~/Games` which means all the 25GiB worth of Rocket League files are there even after the reboot. However, upon running `rocket-league` in the terminal, it asks for installation like it did when I originally installed it. I'm guessing there were some state files that got nuked on the reboot?

Here's what it looks like:
```
[Core] INFO: Trying to re-use existing login session...
[Core] INFO: Fetching metadata for 142 app(s).
[cli] INFO: Preparing download for "Rocket League®" (Sugar)...
[Core] INFO: Parsing game manifest...
[Core] INFO: Install path: /home/getpsyched/Games/rocket-league/rocketleague
[Core] INFO: Selected CDN: epicgames-download1.akamaized.net (https)
[cli] INFO: Install size: 24295.80 MiB
[cli] INFO: Download size: 21969.58 MiB (Compression savings: 2.0%)
[cli] INFO: Reusable size: 0.00 MiB (chunks) / 0.00 MiB (unchanged / skipped)
[cli] INFO: Downloads are resumable, you can interrupt the download with CTRL-C and resume it using the same command later on.

This game may have compatibility issues or require additional setup, see: https://github.com/CommandMC/EpicLinux/wiki/Rocket-League:-BakkesMod

Do you wish to install "Rocket League®"? [Y/n]:
```

I don't want to continue this out of fear that it will start overwriting what I already have and I'll have to wait another 6 hours. Can you help, @fufexan?

I have also tried running:
```sh
PATH="/nix/store/zby4wd8fpvmx1z162449ds77hb46znah-wine-tkg-full-wayland-8.8/bin:$PATH"
WINEPREFIX=~/Games/rocket-league/ nvidia-offload wine ~/Games/rocket-league/rocketleague/Binaries/Win64/RocketLeague.exe
```
The game launches with a black screen, goes full screen and then crashes. I have this log here: [rocket-league-launch.log](https://github.com/fufexan/nix-gaming/files/11584343/rl2.log)

Originally, when it auto-launched after install, it had crashed with the following log: [backtrace.txt](https://github.com/fufexan/nix-gaming/files/11584368/backtrace.txt)

**Comment by fufexan** [comment, marked as answer]
That's really weird. All I know is that the state of the game is also saved in `~/Documents/My Games/Rocket League`. There are also binaries in there as far as I know, so you may want to also have that excluded from wipe.

**Reply by GetPsyched** [reply]
I have excluded that from the wipe and that made it so that the game launches (originally this is why it asked for install again). But it errors out and crashes anyways.
Here's the error:
```
err:   DXGI: EnterFullscreenMode: Cannot query containing output
0134:fixme:faultrep:ReportFault 00000000005FC3A0 0x0 stub
0134:err:unwind:RtlUnwindEx invalid frame 0000000000600008 (0000000000132000-0000000000600000)
0134:err:seh:call_stack_handlers invalid frame 0000000000600088 (0000000000132000-0000000000600000)
0134:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
```

**Reply by fufexan** [reply]
Try using a different wine, for example `wine-ge`.

**Reply by GetPsyched** [reply]
How do I do that? Since I don't have wine declared anywhere separately but by this repo itself.

**Reply by fufexan** [reply]
You can override `rocket-league` like this:
```nix
(rocket-league.override {wine = inputs.nix-gaming.packages.${pkgs.system}.wine-ge;})
```

**Reply by GetPsyched** [reply]
It worked without this. I was running using `steam-run rocket-league` instead of `gamemoderun rocket-league`. However the game keeps crashing while in game. Any reason for this? I don't see any reason for the crash in the terminal.

It is now just crashing at this screen:
![image](https://github.com/fufexan/nix-gaming/assets/43472218/1049ebdc-ebac-4ea9-938e-54b1e046eb60)

**Reply by GetPsyched** [reply]
So, apparently the game not loading _was_ because of the normal wine. Overriding it worked. The game crashes sometimes but I have no clue why. Other than that, I'm getting consistent 330 FPS
